realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -30,20 +32,69 @@ RealSenseNodeFactory::RealSenseNodeFactory()
     rs2::log_to_console(severity);
 }
 
-void RealSenseNodeFactory::onInit()
+bool RealSenseNodeFactory::deviceMatches(rs2::device& dev, std::string& serial_no, std::string& usb_port_id)
 {
-    try{
-        auto list = _ctx.query_devices();
-        if (0 == list.size())
+    ROS_DEBUG("Port ID: %s", dev.get_info(RS2_CAMERA_INFO_PHYSICAL_PORT));
+
+    std::string serial_number = dev.get_info(RS2_CAMERA_INFO_SERIAL_NUMBER);
+
+    std::string port_id = parseUsbPortId(dev.get_info(RS2_CAMERA_INFO_PHYSICAL_PORT));
+
+    if (usb_port_id.empty())
```

Comment:
This could be move up since it does not involve comparing to the actual device port id.